src/dataworker/Dataworker.ts

```diff
@@ -106,13 +110,42 @@ export class Dataworker {
     };
   }
 
-  async buildSlowRelayRoot(bundleBlockNumbers: BundleEvaluationBlockNumbers) {
-    this._loadData();
-
-    // For each destination chain ID key in unfilledDeposits
-    //     Order by fillAmountRemaining
-    //     Make Leaf for fill
-    // Construct root
+  async buildSlowRelayRoot(bundleBlockNumbers: BundleEvaluationBlockNumbers): Promise<MerkleTree<RelayData>> | null {
+    const { unfilledDeposits } = this._loadData();
+    // TODO: Use `bundleBlockNumbers` to decide how to filter which blocks to keep in `unfilledDeposits`.
+
+    if (Object.keys(unfilledDeposits).length === 0) return null;
+
+    const leaves: RelayData[] = [];
+    Object.keys(unfilledDeposits).forEach((destinationChainId) => {
+      leaves.push(
+        ...unfilledDeposits[destinationChainId].map((deposit) => {
+          return {
+            depositor: deposit.deposit.depositor,
+            recipient: deposit.deposit.recipient,
+            destinationToken: deposit.deposit.depositor,
+            amount: deposit.deposit.amount.toString(),
+            originChainId: deposit.deposit.originChainId.toString(),
+            destinationChainId: deposit.deposit.destinationChainId.toString(),
+            realizedLpFeePct: deposit.deposit.realizedLpFeePct.toString(),
+            relayerFeePct: deposit.deposit.relayerFeePct.toString(),
+            depositId: deposit.deposit.depositId.toString(),
+          };
+        })
```

Comment:
for this you can simply cast the deposit to a `RelayData` like this:
```suggestion
        ...unfilledDeposits[destinationChainId].map((deposit) => deposit as DepositData)
```

Reply:
As discussed offline, typescript casting is really a type assertion and doesn't change the object at runtime https://stackoverflow.com/questions/52296152/typescript-cast-using-as